lib/engine/game/g_1836_jr56/game.rb

```diff
@@ -27,130 +27,164 @@ def cert_limit(_player = nil)
-                  }].freeze
+        def game_phases
+          phase_list = [
+            {
+              name: '2',
+              train_limit: 4,
+              tiles: [:yellow],
+              status: %w[escrow facing_2],
+              operating_rounds: 1,
+            },
+            {
+              name: "2'",
+              on: "2'",
+              train_limit: 4,
+              tiles: [:yellow],
+              status: %w[escrow facing_3],
+              operating_rounds: 1,
+            },
+            {
+              name: '3',
+              on: '3',
+              train_limit: 4,
+              tiles: %i[yellow green],
+              operating_rounds: 2,
+              status: %w[escrow facing_3 can_buy_companies],
+            },
+            {
+              name: "3'",
+              on: "3'",
+              train_limit: 4,
+              tiles: %i[yellow green],
+              operating_rounds: 2,
+              status: %w[escrow facing_4 can_buy_companies],
+            },
+            {
+              name: '4',
+              on: '4',
+              train_limit: 3,
+              tiles: %i[yellow green],
+              operating_rounds: 2,
+              status: %w[escrow facing_4 can_buy_companies],
+            },
+            {
+              name: "4'",
+              on: "4'",
+              train_limit: 3,
+              tiles: %i[yellow green],
+              operating_rounds: 2,
+              status: %w[incremental facing_5 can_buy_companies],
+            },
+            {
+              name: '5',
+              on: '5',
+              train_limit: 2,
+              tiles: %i[yellow green brown],
+              status: %w[incremental facing_5],
+              operating_rounds: 3,
+            },
+            {
+              name: "5'",
+              on: "5'",
+              train_limit: 2,
+              tiles: %i[yellow green brown],
+              status: %w[fullcap facing_6],
+              operating_rounds: 3,
+            },
+            {
+              name: '6',
+              on: '6',
+              train_limit: 2,
+              tiles: %i[yellow green brown gray],
+              status: %w[fullcap facing_6 upgradable_towns no_loans],
+              operating_rounds: 3,
+            },
+            {
+              name: '8',
+              on: '8',
+              train_limit: 2,
+              tiles: %i[yellow green brown gray black],
+              status: %w[fullcap facing_6 upgradable_towns no_loans],
+              operating_rounds: 3,
+            },
+            {
+              name: 'D',
+              on: 'D',
+              train_limit: 2,
+              tiles: %i[yellow green brown gray],
+              status: %w[fullcap facing_6 upgradable_towns no_loans],
+              operating_rounds: 3,
+            },
+          ]
+          phase_list.reject! { |p| p[:name] == 'D' } unless diesel_variant?
```

Comment:
change this to
diesel_variant? reject! : reject!

Reply:
I'm not sure what you mean here. Could you elaborate?

Reply:
@tobymao could you explain what you're requesting here? I'm not understanding.